quine-cypher/src/test/scala/com/thatdot/quine/VariableLengthRelationshipPattern.scala

```diff
@@ -438,4 +438,25 @@ class VariableLengthRelationshipPatternMatrix extends CypherHarness("variable-le
       expectedCanContainAllNodeScan = true
     )
   }
+
+  describe("variable length relationships with constraints") {
+    testQuery(
+      "MATCH (a)-[*1]->(b)-->(c) RETURN a.name, c.name",
```

Comment:
Thanks so much for this!
These were compiling to the above `expressions.NoneIterablePredicate` case with e.g.
```
list: Variable(  REL9(9a6dfa28-fbf8-4612-adfd-2f2e04c8580d))
variable: Variable(  REL9(9a6dfa28-fbf8-4612-adfd-2f2e04c8580d))
pred: Equals(Variable(  REL19(346e9c84-8bf9-4982-830a-42ba215968d6)),Variable(  REL9(9a6dfa28-fbf8-4612-adfd-2f2e04c8580d)))
```
which is something like `none(REL9 in REL9 WHERE REL19 = REL9)`, which would blow trying to add the "variable"  `REL9` because it was already in scope as the name of the thing being iterated over.

I tried reproducing that writing expressions like `MATCH (a)-->(b) WHERE a.age = 10 AND none(a in [a] WHERE a.age = a) RETURN b`
but that just generated a fresh `a` as the variable:
```
list: ListLiteral(List(Variable(  a@7)))
variable: Variable(  a@42)
pred: In(Property(Variable(  a@42),PropertyKeyName(age)),ListLiteral(List(Variable(  a@42))))
```